src/haystack_pydoc_tools/renderers.py

```diff
@@ -145,3 +148,65 @@ def _doc_version(self) -> str:
         Returns the hardcoded docs version 2.0.
         """
         return "v2.0"
+
+
+@dataclasses.dataclass
+class ReadmeCoreRenderer(ReadmeRenderer):
+    """
+    This custom Renderer behaves just like the ReadmeRenderer but gets the version from `hatch`.
+    This is meant to be used by the Haystack core repository.
+    """
+
+    def _doc_version(self) -> str:
+        """
+        Returns the docs version.
+        """
+        # We're assuming hatch is installed and working
+        res = subprocess.run(["hatch", "version"], capture_output=True, check=True)
+        res.check_returncode()
+        full_version = res.stdout.decode().strip()
+        major, minor = full_version.split(".")[:2]
+        if "rc0" in full_version:
```

Comment:
Maybe `if "-rc" in full_version:`? Shouldn't be true also for RC1 and the likes?

Reply:
Can't do. After the [`minor_version_release.yml` workflow](https://github.com/deepset-ai/haystack/blob/main/.github/workflows/minor_version_release.yml) runs we update the current unstable to remove the `-unstable` suffix.